modules/standard/Random.chpl

```diff
@@ -1341,6 +1705,31 @@ module Random {
     }
   }
 
+  // Do a binary search of 'a' for the first occurrence of 'x'
+  // returns the index of the first occurrence of 'x' in 'a' between 'start' and 'stop'
+  // or the index where 'x' should be inserted to maintain sorted order
+  private proc binarySearchFirst(
```

Comment:
Can we leave a `todo` here about deciding whether to add these to the Search module itself instead of implementing them here?